src/main/scala/org/apache/mesos/chronos/scheduler/jobs/Schedule.scala

```diff
@@ -1,44 +1,162 @@
     if (recurrences.exists(_ == 0)) None
     else {
       val nextOffset = offset + 1
-      val nextInvocationTime = Schedule.addPeriods(originTime, period, nextOffset.asInstanceOf[Int] /* jodatime doesn't support longs in their plus methods. What to do about overflow? */)
+      val nextInvocationTime = Schedules.addPeriods(originTime, period, nextOffset.asInstanceOf[Int] /* jodatime doesn't support longs in their plus methods. What to do about overflow? */)
 
-      Some(Schedule(schedule, scheduleTimeZone, nextInvocationTime, originTime, nextOffset, recurrences.map(_ - 1), period))
+      Some(Iso8601Schedule(schedule, scheduleTimeZone, nextInvocationTime, originTime, nextOffset, recurrences.map(_ - 1), period))
     }
   }
 
-  def toZeroOffsetISO8601Representation: String = {
+  def toStringRepresentation: String = {
     Iso8601Expressions.create(recurrences.getOrElse(-1), invocationTime, period)
   }
 }
 
-object Schedule {
-  def parse(scheduleStr: String, timeZoneStr: String = ""): Option[Schedule] = {
+/**
+ * A Unix Cron schedule
+ *
+ * @param schedule the text representation of this schedule
+ * @param scheduleTimeZone the time zone the schedule should be executed from
+ * @param invocationTime the time to execute this schedule (UTC)
+ * @param lastExecutionTime the last time this schedule was run (UTC)
+ * @param cron the parsed Cron object that is used to perform execution time planning
+ */
+case class CronSchedule(@JsonProperty schedule: String,
+                        @JsonProperty scheduleTimeZone: String,
+                        @JsonProperty invocationTime: DateTime,
+                        @JsonProperty lastExecutionTime: DateTime,
+                         cron: Cron) extends Schedule {
+  type SType = CronSchedule
+  val recurrences = None
+  val scheduleType = CronType
+
+  lazy val next: Option[CronSchedule] = {
+    val tz = if(scheduleTimeZone != "") DateTimeZone.forID(scheduleTimeZone) else DateTimeZone.UTC
+    val currentTimeUserTZ = invocationTime.withZone(tz)
+    val nextExecUserTZ = ExecutionTime.forCron(cron).nextExecution(currentTimeUserTZ)
+    val nextExec = nextExecUserTZ.withZone(DateTimeZone.UTC)
+
+    Some(copy(invocationTime = nextExec, lastExecutionTime = invocationTime))
+  }
+
+  def toStringRepresentation = cron.asString()
+}
+
+object Schedules {
+  private val log = Logger.getLogger(getClass.getName)
+  private val cronDefinition = CronDefinitionBuilder.instanceDefinitionFor(CronTypeEnum.UNIX)
+
+  /**
+   * Parses a text representation of a schedule into a Schedule object, or None if the parsing fails
+   * @param scheduleStr the string representation of the schedule
+   * @param timeZoneStr the time zone to execute the schedule in
+   * @param scheduleType the type of schedule (Iso8601Type or CronType)
+   * @param currentTime the current DateTime in UTC
+   * @return
+   */
+  def parse(scheduleStr: String,
+            timeZoneStr: String = "",
```

Comment:
Does this need to be an empty string, or would it be cleaner to make this an `Option`?